src/v/storage/compaction_reducers.cc

```diff
@@ -229,37 +229,34 @@ copy_data_segment_reducer::filter(model::record_batch&& batch) {
       h, std::move(ret), model::record_batch::tag_ctor_ng{});
     return new_batch;
 }
+
 ss::future<ss::stop_iteration> copy_data_segment_reducer::do_compaction(
   model::compression original, model::record_batch&& b) {
     using stop_t = ss::stop_iteration;
     auto to_copy = filter(std::move(b));
     if (to_copy == std::nullopt) {
         co_return stop_t::no;
     }
-    auto rb = co_await compress_batch(original, std::move(to_copy.value()));
-    co_await ss::do_with(std::move(rb), [this](model::record_batch& batch) {
-        auto const start_offset = _appender->file_byte_offset();
-        auto const header_size = batch.header().size_bytes;
-        _acc += header_size;
-        if (_idx.maybe_index(
-              _acc,
-              32_KiB,
-              start_offset,
-              batch.base_offset(),
-              batch.last_offset(),
-              batch.header().first_timestamp,
-              batch.header().max_timestamp)) {
-            _acc = 0;
-        }
-        return storage::write(*_appender, batch)
-          .then([this, start_offset, header_size] {
-              vassert(
-                _appender->file_byte_offset() == start_offset + header_size,
-                "Size must be deterministic. Expected:{} == {}",
-                _appender->file_byte_offset(),
-                start_offset + header_size);
-          });
-    });
+    auto batch = co_await compress_batch(original, std::move(to_copy.value()));
```

Comment:
unless you want to suppress clang-format in the previous commit so that the indentation makes the changes really clear, I'd squash this commit into the previous one. I'd probably vote to squash the changes are pretty straight forward.